### ISSUE TITLE
[1.60] Thaumcraft aspects not working in JEI

### ISSUE THREAD

**Issue by eugestarr** [NillerMedDild/Enigmatica2Expert#1092]
**Modpack version**
1.60

**Can the issue be reproduced?**
Yes

**Logs**
N/A

**Issue**
JEI not showing a tab with the list of items where a Thaumcraft aspect can be acquired

**Comment by NillerMedDild** [comment]
Thank you for the report, I will investigate the issue.

**Comment by Frikkle** [comment]
I can confirm that the Thaumcraft aspects tab does not show properly for me either. Seems like an issue similar to #1024.

My logs report nothing about thaumicjei though.

**Comment by nobleobject** [comment]
Can also confirm the same issue. tried re creating the json file to no avail as well as regenerating the config through setting alwaysRecreateAspectFromItemStackFile to true.

**Comment by N00bSkyw4lk3r** [comment]
Sadly the 1.61 Beta hotfix did not solve this issue. I just wanted to dive into the "new" thaumcraft and feeling kinda lost now, not knowing how to best get all the aspects from - I guess ill have to downgrade to 1.59a for this one.
Im not much of a programmer, but i looked through the "latest log". This is what i found about the thaumic jei, though i cannot tell how much of use this would be:
`[14:55:22] [Client thread/ERROR]: Failed to register mod plugin: class com.buuz135.thaumicjei.ThaumcraftJEIPlugin
java.lang.ClassCastException: net.minecraft.item.ItemAir cannot be cast to crazypants.enderio.base.power.forge.item.IInternalPoweredItem
	at crazypants.enderio.base.power.forge.item.InternalPoweredItemCap.<init>(InternalPoweredItemCap.java:15) ~[InternalPoweredItemCap.class:?]
	at crazypants.enderio.powertools.machine.capbank.BlockItemCapBank.initCapabilities(BlockItemCapBank.java:22) ~[BlockItemCapBank.class:?]
	at net.minecraft.item.ItemStack.forgeInit(ItemStack.java:1215) ~[aip.class:?]
	at net.minecraft.item.ItemStack.<init>(ItemStack.java:138) ~[aip.class:?]
	at com.buuz135.thaumicjei.ThaumcraftJEIPlugin.lambda$register$5(ThaumcraftJEIPlugin.java:156) ~[ThaumcraftJEIPlugin.class:?]
	at com.buuz135.thaumicjei.ThaumcraftJEIPlugin$$Lambda$5604/626833317.apply(Unknown Source) ~[?:?]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193) ~[?:1.8.0_51]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175) ~[?:1.8.0_51]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193) ~[?:1.8.0_51]
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374) ~[?:1.8.0_51]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:512) ~[?:1.8.0_51]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:502) ~[?:1.8.0_51]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708) ~[?:1.8.0_51]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:1.8.0_51]
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499) ~[?:1.8.0_51]
	at com.buuz135.thaumicjei.ThaumcraftJEIPlugin.register(ThaumcraftJEIPlugin.java:159) ~[ThaumcraftJEIPlugin.class:?]
	at mezz.jei.startup.JeiStarter.registerPlugins(JeiStarter.java:202) [JeiStarter.class:?]
	at mezz.jei.startup.JeiStarter.start(JeiStarter.java:73) [JeiStarter.class:?]
	at mezz.jei.startup.ProxyCommonClient.loadComplete(ProxyCommonClient.java:136) [ProxyCommonClient.class:?]
	at mezz.jei.JustEnoughItems.loadComplete(JustEnoughItems.java:55) [JustEnoughItems.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_51]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_51]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_51]
	at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_51]
	at net.minecraftforge.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:624) [FMLModContainer.class:?]
	at sun.reflect.GeneratedMethodAccessor2.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_51]
	at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_51]
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91) [guava-21.0.jar:?]
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150) [guava-21.0.jar:?]
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76) [guava-21.0.jar:?]
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) [guava-21.0.jar:?]
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71) [guava-21.0.jar:?]
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116) [guava-21.0.jar:?]
	at com.google.common.eventbus.EventBus.post(EventBus.java:217) [guava-21.0.jar:?]
	at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:219) [LoadController.class:?]
	at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:197) [LoadController.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_51]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_51]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_51]
	at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_51]
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91) [guava-21.0.jar:?]
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150) [guava-21.0.jar:?]
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76) [guava-21.0.jar:?]
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) [guava-21.0.jar:?]
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71) [guava-21.0.jar:?]
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116) [guava-21.0.jar:?]
	at com.google.common.eventbus.EventBus.post(EventBus.java:217) [guava-21.0.jar:?]
	at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:136) [LoadController.class:?]
	at net.minecraftforge.fml.common.Loader.initializeMods(Loader.java:752) [Loader.class:?]
	at net.minecraftforge.fml.client.FMLClientHandler.finishMinecraftLoading(FMLClientHandler.java:336) [FMLClientHandler.class:?]
	at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:535) [bib.class:?]
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:378) [bib.class:?]
	at net.minecraft.client.main.Main.main(SourceFile:123) [Main.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_51]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_51]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_51]
	at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_51]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]`

**Comment by N00bSkyw4lk3r** [comment]
Sorry for the Spam-post...i dont know how to post the log file properly, and i do not have the time to figure it out. Feel free to do what you need to do with this post ;)
kr
a n00b

**Comment by NillerMedDild** [comment]
I know the new update didn't fix this issue, that's why it's still open.

**Comment by mitchej123** [comment]
https://github.com/Buuz135/Thaumic-JEI/issues/50

**Comment by Mormoran** [comment]
Keep in mind that JEI (Thaumic Aspects for JEI actually) will not show aspects in JEI or in the recipes (hitting R on something) UNTIL you unlock the relevant knowledge in the Thaumonomicon (alchemy gate, I think). So if someone progresses a lot at once without reloading their game in between unlocking alchemy and doing a lot of progress, the game will feel bugged. It just happened to me.

**Comment by Frikkle** [comment]
> Keep in mind that JEI (Thaumic Aspects for JEI actually) will not show aspects in JEI or in the recipes (hitting R on something) UNTIL you unlock the relevant knowledge in the Thaumonomicon (alchemy gate, I think). So if someone progresses a lot at once without reloading their game in between unlocking alchemy and doing a lot of progress, the game will feel bugged. It just happened to me.

For me the Thaumic JEI aspects tab was showing properly for everyone on our server in 1.59 and disappeared in 1.60. So this seems to be a mod/compatibility issue rather than not having progressed far enough.

**Comment by Mormoran** [comment]
Yes, I now know what you mean, I progressed far enough, and while I do get a cauldron tab in my JEI, I can't see which items give which type of essentia. I've edited my post above. It's proven to be very irritating having to mouse over lots of stuff looking for the right icon, or alt+tabbing to guides with a list of essentia on items.

Hopefully this can get fixed soon, I'm quite enjoying Thaumcraft 6.
